docs/reference/performance.md

```diff
@@ -0,0 +1,212 @@
+Cluster node counts: 11, 62, 307
+Node Type: Standard_D4s_v3 4vCPUs 16Gib memory. Details [here](https://learn.microsoft.com/en-us/azure/virtual-machines/dv3-dsv3-series#dsv3-series)
+
+#### Registry Region Size
+
+ACR has different region sizes. The size of the region determines the max RPS for a single ACR from a single IP. The test focuses on 4 regions of varying sizes. 
+
+- East US 2: Large
+- East US 2 Availability-Zone-enabled: Medium
+- Central US Availability-Zone-enabled: Small
+- East US 2 Canary: Extra Small
+
+ACR regions were selected by size and proximity to the cluster region of East US 2 to minimize cross-region latency. 
+
+#### Workload Resource Type
+
+The test focused on Deployment and Job workloads. Most tests were conducted with a single Deployment with varying replica counts. Job tests applied 5 Jobs simultaneously splitting total pod count evenly across each job. 
+
+#### Distributed caching
+Majority of tests enabled distributed caching across 3 replicas for optimal performance configuration. A subset tested Ratify with 3 replicas with the distributed cache disabled. Distributed caching is critical for reducing duplicate registry requests. It is also essential for external plugins utilizing caching.
+
+Test enabled/disabled RATIFY_EXPERIMENTAL_HIGH_AVAILABILITY feature flag
+
+#### Ratify Single Instance
+This test aimed to determine the workload threshold for a single instance of Ratify before over-exhaustion. Single instance vs 3 Ratify Replicas.
+
+
+### Measurements
+
+1. Registry Throttling
+    - \# of 429 http responses returned from registry
+    - This indicates supported registry RPS in a particular region.
+1. Ratify pod(s) CPU/memory usage: High Memory consumption is correlated with high RPS load on Ratify.
+1. Approximate max verification request duration: Helpful to understand initial non-cached request durations
+3. P95 verification request duration: Shows distribution of request durations through duration of the test. 
+4. Approximate max mutation request duration: Helpful to understand initial non-cache request durations.
+5. P95 verification request duration: Shows distribution of request duration throughout the duration of test. 
+6. \# Ratify Container Restarts: early indication of resource exhaustion and blocking errors at scale
+7. \# Ratify Pod OOMKilled: definite signal of resource over-exhaustion
+
+
+### Data Matrix
+
+Each scenario is grouped by color. Red boxes indicate threshold violations. 
+
+![](../imgs/ratify_perf_results_sept2023.png)
+
+### Test Scenarios Explained
+
+#### Deployments across ACR Regions (Blue)
+
+The results confirmed the hypothesis that as the ACR region size decreases, the registry throttling increases leading to more Ratify timeouts. As pod count increases, the effects of throttling worsen especially in the XS region for ACR. (Note: the extremely large timeout count for 10k pods in XS region is linked to an issue with the HA setup. More investigation required) 
```

Comment:
I wonder if we need to test different ACR regions with the same size, espeically the case that AKS and ACR are in different/same regions.

Reply:
yeah it's definitely a valid point considering latency adds up with the number of serial requests we have. I think in general guidance is that resources should in the same region/close which is why the testing focused on this.